api/src/main/java/org/jmisb/api/klv/KlvParser.java

```diff
@@ -13,6 +17,73 @@ public class KlvParser {
 
     private KlvParser() {}
 
+    /**
+     * Parse an InputStream containing one or more {@link IMisbMessage}s.
+     *
+     * <p>This is an additional interface for parsing KLV metadata. It assumes that {@code is}
+     * contains one or more top-level messages, i.e., byte sequences starting with a Universal Label
+     * (UL). If a particular UL is unsupported it will be returned as a {@link RawMisbMessage}.
+     *
+     * <p>The supported UL are determined by the {@link MisbMessageFactory} singleton.
+     *
+     * @param is The input stream
+     * @param handler The resultant {@link IMisbMessage} objects streamed
+     * @param exceptionHandler The {@link KlvParseException} errors detected in the stream.
+     */
+    public static void parseStream(
+            InputStream is,
+            Consumer<IMisbMessage> handler,
+            Consumer<KlvParseException> exceptionHandler)
+            throws KlvParseException {
+
+        // reusable key array to minimize garbage
+        byte[] key = new byte[UniversalLabel.LENGTH];
+
+        try {
+            while (true) {
+                ByteArrayOutputStream out = new ByteArrayOutputStream();
+
+                // Read the UniversalLabel
+                int read = is.read(key, 0, key.length);
+                if (read < 0) {
+                    break;
+                }
+                if (read != key.length) {
+                    throw new KlvParseException(
```

Comment:
Did you want to use the exceptionHandler consumer here instead?

Reply:
I don't, if the stream ran out or got disconnected... continuing when the InputStream ran out of bytes will lead to parseStream parsing invalid data, if any at all.